common/lib/dependabot/clients/bitbucket.rb

```diff
@@ -52,6 +68,92 @@ def fetch_file_contents(repo, commit, path)
         response.body
       end
 
+      def commits(repo, branch_name = nil)
+        commits_path = "#{repo}/commits/" + branch_name.to_s
+        commits_path += "?pagelen=100"
+
+        response = get(base_url + commits_path)
+
+        JSON.parse(response.body).fetch("values")
+      end
+
+      def branch(repo, branch_name)
+        branch_path = "#{repo}/refs/branches/#{branch_name}"
+        response = get(base_url + branch_path)
+
+        JSON.parse(response.body)
+      end
+
+      def pull_requests(repo, source_branch, target_branch)
+        pr_path = "#{repo}/pullrequests"
+        pr_path += "?status=OPEN&status=MERGED&status=DECLINED&status=SUPERSEDED"
+        next_page = base_url + pr_path
+        pull_requests = []
+        loop do
+          response = get(next_page)
+          result = JSON.parse(response.body)
+          pull_requests.concat result.fetch("values")
+          break unless result.key?("next")
+
+          next_page = result.fetch("next")
+        end
+
+        pull_requests unless source_branch && target_branch
```

Comment:
Are we supposed to bail if either `source_branch` or `target_branch` is `nil`? Or should we let the provider determine defaults for us if not specified in the client?